From c2f6b78eb42f94dd6fb355d04d9e4dbc6be91022 Mon Sep 17 00:00:00 2001
From: BuildTools <unconfigured@null.spigotmc.org>
Date: Thu, 26 Dec 2019 23:08:14 +0000
Subject: [PATCH] Limit TickListServer


diff --git a/src/main/java/net/minecraft/server/TickListServer.java b/src/main/java/net/minecraft/server/TickListServer.java
index 6571fc5..d99c22f 100644
--- a/src/main/java/net/minecraft/server/TickListServer.java
+++ b/src/main/java/net/minecraft/server/TickListServer.java
@@ -53,7 +53,13 @@ public class TickListServer<T> implements TickList<T> {
                 }
                 // CraftBukkit end
             }
-
+            
+            // FeatherSpigot start
+            if (i > 10000) {
+                i = 10000;
+            }
+            // FeatherSpigot end
+            
             this.f.methodProfiler.enter("cleaning");
             timingCleanup.startTiming(); // Paper
             NextTickListEntry<T> nextticklistentry; // CraftBukkit - decompile error
-- 
2.8.1.windows.1

